test/app/core/managers/audio_record_services_test.dart

```diff
@@ -0,0 +1,313 @@
+      },
+    );
+
+    test(
+      'start() do not crash if soundRecorder crash',
+      () async {
+        // arrange
+        final mockPermissionHandlerPlatform =
+            PermissionHandlerPlatform.instance;
+        debugDefaultTargetPlatformOverride = TargetPlatform.iOS;
+
+        when((() => mockPermissionHandlerPlatform
+                .checkPermissionStatus(Permission.microphone)))
+            .thenAnswer((_) async => PermissionStatus.granted);
+
+        when(() => soundRecorder.isStopped).thenReturn(true);
+        when(() => soundRecorder.openRecorder()).thenAnswer((_) async => null);
+        when(() => soundRecorder.setSubscriptionDuration(any()))
+            .thenThrow(Exception());
+
+        when(() => audioSyncManager.audioFile(
+              session: any(named: 'session'),
+              sequence: any(named: 'sequence'),
+            )).thenAnswer((_) async => '/tmp/file_name.txt');
+
+        // act / assert
+        await expectLater(
+          recordServices.start(),
+          completion(isNull),
+        );
+      },
+    );
+
+    test(
+      'stop() close the audio record session',
+      () async {
+        // arrange
+        when(() => soundRecorder.isStopped).thenReturn(false);
+        when(() => soundRecorder.stopRecorder()).thenAnswer((_) async => 'foo');
+        when(() => audioSyncManager.syncAudio()).thenAnswer((_) async => true);
+
+        // act
+        await recordServices.stop();
+
+        // assert
+        verify(() => soundRecorder.isStopped).called(1);
+        verify(() => soundRecorder.stopRecorder()).called(1);
+      },
+    );
+
+    test(
+      'stop() do not crash if soundRecorder crash',
+      () async {
+        // arrange
+        when(() => soundRecorder.isStopped).thenReturn(false);
+        when(() => soundRecorder.stopRecorder()).thenThrow(Exception());
+
+        // act / assert
+        await expectLater(
+          recordServices.stop(),
+          completion(isNull),
+        );
+      },
+    );
+
+    test(
+      'onProgress shows the record progress',
+      () async {
+        // arrange
+        final mockPermissionHandlerPlatform =
+            PermissionHandlerPlatform.instance;
+        debugDefaultTargetPlatformOverride = TargetPlatform.iOS;
+
+        when((() => mockPermissionHandlerPlatform
+                .checkPermissionStatus(Permission.microphone)))
+            .thenAnswer((_) async => PermissionStatus.granted);
+
+        when(() => soundRecorder.isStopped).thenReturn(true);
+        when(() => soundRecorder.openRecorder()).thenAnswer((_) async => null);
+        when(() => soundRecorder.setSubscriptionDuration(any()))
+            .thenAnswer((_) async => null);
+        when(() => soundRecorder.startRecorder(
+              codec: any(named: 'codec'),
+              toFile: any(named: 'toFile'),
+              bitRate: any(named: 'bitRate'),
+              sampleRate: any(named: 'sampleRate'),
+            )).thenAnswer((_) async => null);
```

Comment:
O retorno dos métodos `flutterSoundPlayer.openPlayer()`, `flutterSoundPlayer.setSubscriptionDuration()` e `flutterSoundPlayer.startPlayer()` é `Future<void>`, nunca retornará nulo, acha que faria sentido o mock ser `void` também?
```suggestion
        when(() => soundRecorder.openRecorder()).thenAnswer((_) async {});
        when(() => soundRecorder.setSubscriptionDuration(any()))
            .thenAnswer((_) async {});
        when(() => soundRecorder.startRecorder(
              codec: any(named: 'codec'),
              toFile: any(named: 'toFile'),
              bitRate: any(named: 'bitRate'),
              sampleRate: any(named: 'sampleRate'),
            )).thenAnswer((_) async {});
```